### ISSUE TITLE
State importance of favoring algorithms

### ISSUE THREAD

**Issue by janEbert** [deeplearning540/deeplearning540.github.io#45]
[Lesson 5, question 1](https://github.com/deeplearning540/deeplearning540.github.io/blame/4a13e12503b029e8abc06bb1f499f89399e4142a/source/lesson05/content.rst#L34):
> Which of the following would you apply Deep Learning to?
> [...]
> 2. Calculating the median and interquartile range of a dataset.

[Answer](https://github.com/deeplearning540/deeplearning540.github.io/blame/4a13e12503b029e8abc06bb1f499f89399e4142a/source/lesson05/content.rst#L47):
> 2. no, even though it is a regression problem, you'd need tons of training data and still be off by quite a bit

The answer misses one of the most important points: there are existing, super efficient and 100 %-accurate algorithms for this. :)

**Comment by psteinb** [comment]
Ok, I guess I have to reformulate this. What I mean was to try to learn arithmetic operations using Deep Learning.

**Comment by janEbert** [comment]
I'd say the point still stands – there are already algorithms for arithmetic operations which don't require
- the whole bother of training a neural network, and
- an expensive neural network evaluation that
- only returns approximate results.

**Comment by psteinb** [comment]
I think there is a misunderstanding here ... that is the sole point, that learners discover that quite some tasks (like arithmetic operations) do not make sense to solve with a neural network. ;-)

**Comment by janEbert** [comment]
Yes, but the answer only explains that the problem may be hard with a neural network. It doesn't mention that it's simply not the right tool for the job. :D

That's what I'm getting at; the question is great to get people thinking, but the answer smudges the more important lesson (the task being one that ML _should not be applied_ to) by arguing that neural network training is hard.
At least, that's the impression I get when reading the answer. Could also be that you _want_ the students to think about this and come to the conclusion on their own; in that case, disregard everything I said. :)

**Comment by psteinb** [comment]
I take that point. Could you please send a PR to fix that? ;-)
